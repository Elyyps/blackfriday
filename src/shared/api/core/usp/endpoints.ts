import { uniqueSellingPoints } from "./generate-dummy-data";
import { USP } from "./usp";

export const generateUSP = (): USP => uniqueSellingPoints;
