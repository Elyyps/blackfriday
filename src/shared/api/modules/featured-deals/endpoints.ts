import { dummyFeaturedDealsData } from "./generate-dummy-data";

export const getFeaturedDealsDummyData = () => dummyFeaturedDealsData;
