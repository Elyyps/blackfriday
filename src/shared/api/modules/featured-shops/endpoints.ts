import { dummyFeaturedShopsData } from "./generate-dummy-data";

export const getFeaturedShopsDummyData = () => dummyFeaturedShopsData;
