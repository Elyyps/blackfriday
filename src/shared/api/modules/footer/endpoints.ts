import { bottomFooter, middleFooter, topFooter } from "./generate-dummy-data";
import { Footer } from "./models/footer.module";

export const getFullfooter = (): Footer => ({
  bottomFooter: bottomFooter(),
  middleFooter: middleFooter(),
  topFooter: topFooter()
});
