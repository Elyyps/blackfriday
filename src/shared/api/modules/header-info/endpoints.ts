import { HeaderInfoDummy } from "./generate-dummy-data";

export const getHeaderInfoDummyData = () => HeaderInfoDummy;
