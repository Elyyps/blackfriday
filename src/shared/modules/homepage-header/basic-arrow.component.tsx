import * as React from "react";
import { IconComponent } from "@app/prep/modules-prep/core";
import { getArrow } from "@app/constants/icons";
import styles from "./homepage-header-component.module.scss";

interface IBasicArrowProps {
  className?: string;
  customArrow?: JSX.Element;
  onClick?: () => void;
  prev?: boolean;
}

const BasicArrow = (props: IBasicArrowProps) => {
  const { className, onClick } = props;
  let result;
  const customStyle = "homepage-header__slick-arrow-" + `${props.prev ? "prev" : "next"}`;
  if (props.customArrow === undefined) {
    result = (
      <button
        className={`${className} ${styles["homepage-header__slick-arrow"]} ${styles[customStyle]}`}
        style={{ display: "block" }}
        onClick={onClick}
      >
        <IconComponent icon={getArrow(props.prev)} size={"7px"} />
      </button>
    );
  } else {
    result = (
      <div className={className} onClick={onClick} role="button" aria-label="nextButton">
        {props.customArrow}
      </div>
    );
  }

  return result;
};
export { BasicArrow };
