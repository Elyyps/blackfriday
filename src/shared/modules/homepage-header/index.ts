export * from "./components/homepage-header.component";
