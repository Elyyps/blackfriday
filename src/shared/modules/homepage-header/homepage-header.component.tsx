import * as React from "react";
import styles from "./homepage-header-component.module.scss";

import Slider, { Settings } from "react-slick";

import { CHEVRON_RIGHT } from "@app/constants/icons";
import { ArrowPanel, LabelComponent } from "@app/core";
import { getHomePageHeaderSetting } from "./slider-settings";
import { LinkComponent } from "@app/core/link";
import { HomepageHeader } from "@app/api/modules/homepage-header/homepage-header.module";

export interface IHomePageHeaderComponentProps {
  customNextArrow?: JSX.Element | undefined;
  customPrevArrow?: JSX.Element | undefined;
  homePageModule: HomepageHeader;
}

const HomePageHeaderComponent = (props: IHomePageHeaderComponentProps) => {
  const settings: Settings = getHomePageHeaderSetting(
    1,
    props.customNextArrow,
    props.customPrevArrow,
    styles["homepage-header__slick-slider"]
  );

  return (
    <div className={styles["homepage-header"]}>
      <div className={styles["homepage-header-top"]}>
        <div className="uk-container">
          <div className="uk-grid uk-child-width-1-2@m">
            <div>
              <div className={styles["homepage-header-wrap"]}>
                <LabelComponent text={"Black Friday 2019: NOG 10 dagen!"} />
                <div className={styles["homepage-header__title"]}>
                  <h1>
                    <strong>{props.homePageModule.title}</strong>
                  </h1>
                </div>
                <ul className={`uk-list-check ${styles["homepage-header__list"]}`}>
                  {props.homePageModule.description.map((text, index) => (
                    <li key={index}>{text}</li>
                  ))}
                </ul>
                <div className={styles["homepage-header__cards"]}>
                  {props.homePageModule.arrowPanelslinks.map((panel, key) => (
                    <ArrowPanel key={key} title={panel.text.title} icon={panel.icon} link={panel.text.url} />
                  ))}
                </div>
              </div>
              <div className={`uk-invert ${styles["homepage-header-info"]}`}>
                <div className={styles["homepage-header-info__col"]}>
                  <div className={styles["homepage-header-info__label"]}>{props.homePageModule.startingDateText}</div>
                  <div className={styles["homepage-header-info__item"]}>
                    <h3>{props.homePageModule.startingDate}</h3>
                  </div>
                </div>
                <div className={styles["homepage-header-info__col"]}>
                  <div className={styles["homepage-header-info__label"]}>
                    {props.homePageModule.durationBlackFridayText}
                  </div>
                  <div className={styles["homepage-header-info__item"]}>
                    <h3>{props.homePageModule.durationBlackFriday}</h3>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <Slider {...settings}>
                {props.homePageModule.slides.map((slide, index) => (
                  <div key={index} className={styles["homepage-header__slide-container"]}>
                    <div
                      style={{ backgroundImage: `url(${slide.image})` }}
                      className={styles["homepage-header__slider-item"]}
                    />
                    <div className={styles["homepage-header__slider-control"]}>
                      <div className={styles["homepage-header__slider-title"]}> {slide.title}</div>
                      <LinkComponent icon={CHEVRON_RIGHT} to={slide.link.url} animated variant={"quaternary"}>
                        {slide.link.title}
                      </LinkComponent>
                    </div>
                  </div>
                ))}
              </Slider>
            </div>
          </div>
        </div>
      </div>
      <div className={styles["homepage-header-bottom"]}>
        <div className={`uk-invert ${styles["homepage-header-bottom__wrap"]}`}>
          <div className="uk-container">
            <div className="uk-grid uk-child-width-1-2@s">
              <div>
                <div className={styles["homepage-header-bottom__label"]}>
                  <h2>{props.homePageModule.popularProductsTitle}</h2>
                </div>
                <div className={styles["homepage-header-bottom__cards"]}>
                  {props.homePageModule.popularProducts.map((product, key) => (
                    <div key={key}>
                      <ArrowPanel title={product.text.title} icon={product.icon} link={product.text.url} />
                    </div>
                  ))}
                </div>
              </div>
              <div>
                <div className={styles["homepage-header-bottom__label"]}>
                  <h2>{props.homePageModule.popularStoresTitle}</h2>
                </div>
                <div className={styles["homepage-header-bottom__cards"]}>
                  {props.homePageModule.popularStores.map((store, key) => (
                    <div key={key}>
                      <ArrowPanel title={store.text.title} icon={store.icon} link={store.text.url} />
                    </div>
                  ))}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export { HomePageHeaderComponent };
