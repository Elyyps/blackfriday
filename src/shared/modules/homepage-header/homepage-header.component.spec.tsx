import React from "react";
import { shallow } from "enzyme";
import { HomePageHeaderComponent } from "../homepage-header";

import { getHeaderpageHeader } from "@app/api/modules/homepage-header/endpoints";
describe("[homepageHeader]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<HomePageHeaderComponent homePageModule={getHeaderpageHeader()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
