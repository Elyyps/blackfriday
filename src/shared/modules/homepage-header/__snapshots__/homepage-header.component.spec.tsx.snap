// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[homepageHeader] should render component without crashing 1`] = `
<div
  className="homepage-header"
>
  <div
    className="homepage-header-top"
  >
    <div
      className="uk-container"
    >
      <div
        className="uk-grid uk-child-width-1-2@m"
      >
        <div>
          <div
            className="homepage-header-wrap"
          >
            <LabelComponent
              text="Black Friday 2019: NOG 10 dagen!"
            />
            <div
              className="homepage-header__title"
            >
              <h1>
                <strong>
                  Black Friday BelgiÃ«
                </strong>
              </h1>
            </div>
            <ul
              className="uk-list-check homepage-header__list"
            >
              <li
                key="0"
              >
                Overzicht van alle Black Friday winkels
              </li>
              <li
                key="1"
              >
                Altijd up-to-date van de laatste deals
              </li>
              <li
                key="2"
              >
                Mis niets van Black Friday 2019
              </li>
            </ul>
            <div
              className="homepage-header__cards"
            >
              <ArrowPanel
                icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/medal.svg"
                key="0"
                link={
                  Object {
                    "title": "Productdeals",
                    "url": "./",
                  }
                }
              />
              <ArrowPanel
                icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/store.svg"
                key="1"
                link={
                  Object {
                    "title": "Deelnemende winkels",
                    "url": "./",
                  }
                }
              />
            </div>
          </div>
          <div
            className="uk-invert homepage-header-info"
          >
            <div
              className="homepage-header-info__col"
            >
              <div
                className="homepage-header-info__label"
              >
                Wanneer is Black Friday
              </div>
              <div
                className="homepage-header-info__item"
              >
                <h3>
                  29 November 2019
                </h3>
              </div>
            </div>
            <div
              className="homepage-header-info__col"
            >
              <div
                className="homepage-header-info__label"
              >
                Hoe lang nog?
              </div>
              <div
                className="homepage-header-info__item"
              >
                <h3>
                  6 maanden & 2 dagen 
                </h3>
              </div>
            </div>
          </div>
        </div>
        <div>
          <Slider
            className="homepage-header__slick-slider"
            dots={false}
            infinite={false}
            nextArrow={<BasicArrow />}
            prevArrow={
              <BasicArrow
                prev={true}
              />
            }
            slidesToScroll={1}
            slidesToShow={1}
            speed={500}
          >
            <div
              className="homepage-header__slide-container"
              key="0"
            >
              <div
                className="homepage-header__slider-item"
                style={
                  Object {
                    "backgroundImage": "url(https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png)",
                  }
                }
              />
              <div
                className="homepage-header__slider-control"
              >
                <div
                  className="homepage-header__slider-title"
                >
                   
                  Handige Black Friday Tips
                </div>
                <LinkComponent
                  animated={true}
                  icon={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "chevron-right.svg",
                    }
                  }
                  to="./"
                  variant="quaternary"
                >
                  Lees meer
                </LinkComponent>
              </div>
            </div>
            <div
              className="homepage-header__slide-container"
              key="1"
            >
              <ModalComponent
                isTriggerOnClick={true}
                trigger={
                  <div
                    className="homepage-header__slider-item homepage-header__slider-item-video"
                    style={
                      Object {
                        "backgroundImage": "url(https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/video-thumbnail.jpg)",
                      }
                    }
                  >
                    <IconComponent
                      icon={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "play-button.svg",
                        }
                      }
                      size="80px"
                    />
                  </div>
                }
                variant="big"
              >
                <ReactPlayer
                  config={
                    Object {
                      "dailymotion": Object {
                        "params": Object {
                          "api": 1,
                          "endscreen-enable": false,
                        },
                        "preload": false,
                      },
                      "facebook": Object {
                        "appId": "1309697205772819",
                      },
                      "file": Object {
                        "attributes": Object {},
                        "dashVersion": "2.9.2",
                        "forceAudio": false,
                        "forceDASH": false,
                        "forceHLS": false,
                        "forceVideo": false,
                        "hlsOptions": Object {},
                        "hlsVersion": "0.10.1",
                        "tracks": Array [],
                      },
                      "mixcloud": Object {
                        "options": Object {
                          "hide_cover": 1,
                        },
                      },
                      "soundcloud": Object {
                        "options": Object {
                          "buying": false,
                          "download": false,
                          "liking": false,
                          "sharing": false,
                          "show_comments": false,
                          "show_playcount": false,
                          "visual": true,
                        },
                      },
                      "twitch": Object {
                        "options": Object {},
                      },
                      "vimeo": Object {
                        "playerOptions": Object {
                          "autopause": false,
                          "byline": false,
                          "portrait": false,
                          "title": false,
                        },
                        "preload": false,
                      },
                      "wistia": Object {
                        "options": Object {},
                      },
                      "youtube": Object {
                        "embedOptions": Object {},
                        "playerVars": Object {
                          "iv_load_policy": 3,
                          "modestbranding": 1,
                          "playsinline": 1,
                          "rel": 0,
                          "showinfo": 0,
                        },
                        "preload": false,
                      },
                    }
                  }
                  controls={true}
                  height="500px"
                  light={false}
                  loop={false}
                  muted={false}
                  onBuffer={[Function]}
                  onBufferEnd={[Function]}
                  onDisablePIP={[Function]}
                  onDuration={[Function]}
                  onEnablePIP={[Function]}
                  onEnded={[Function]}
                  onError={[Function]}
                  onPause={[Function]}
                  onPlay={[Function]}
                  onProgress={[Function]}
                  onReady={[Function]}
                  onSeek={[Function]}
                  onStart={[Function]}
                  pip={false}
                  playbackRate={1}
                  playing={true}
                  playsinline={false}
                  progressInterval={1000}
                  style={Object {}}
                  url="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/BlackFridayNederland-16x9.mp4"
                  volume={null}
                  width="100%"
                  wrapper="div"
                />
              </ModalComponent>
            </div>
            <div
              className="homepage-header__slide-container"
              key="2"
            >
              <div
                className="homepage-header__slider-item"
                style={
                  Object {
                    "backgroundImage": "url(https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png)",
                  }
                }
              />
              <div
                className="homepage-header__slider-control"
              >
                <div
                  className="homepage-header__slider-title"
                >
                   
                  Handige Black Friday Tips
                </div>
                <LinkComponent
                  animated={true}
                  icon={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "chevron-right.svg",
                    }
                  }
                  to="./"
                  variant="quaternary"
                >
                  Lees meer
                </LinkComponent>
              </div>
            </div>
          </Slider>
        </div>
      </div>
    </div>
  </div>
  <div
    className="homepage-header-bottom"
  >
    <div
      className="uk-invert homepage-header-bottom__wrap"
    >
      <div
        className="uk-container"
      >
        <div
          className="uk-grid uk-child-width-1-2@s"
        >
          <div>
            <div
              className="homepage-header-bottom__label"
            >
              <h2>
                Populaire onderwerpen
              </h2>
            </div>
            <div
              className="homepage-header-bottom__cards"
            >
              <div
                key="0"
              >
                <ArrowPanel
                  image="iears.png"
                  key="0"
                  link={
                    Object {
                      "title": "Apple Airpods",
                      "url": "/",
                    }
                  }
                />
              </div>
              <div
                key="1"
              >
                <ArrowPanel
                  image="imac.png"
                  key="1"
                  link={
                    Object {
                      "title": "Apple iPhones",
                      "url": "/",
                    }
                  }
                />
              </div>
              <div
                key="2"
              >
                <ArrowPanel
                  image="iphone.png"
                  key="2"
                  link={
                    Object {
                      "title": "Apple MacBooks",
                      "url": "/",
                    }
                  }
                />
              </div>
              <div
                key="3"
              >
                <ArrowPanel
                  image="loader.png"
                  key="3"
                  link={
                    Object {
                      "title": "Apple iMacs",
                      "url": "/",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div>
            <div
              className="homepage-header-bottom__label"
            >
              <h2>
                Populaire winkels
              </h2>
            </div>
            <div
              className="homepage-header-bottom__cards"
            >
              <div
                key="0"
              >
                <ArrowPanel
                  image="cool-blue.svg"
                  key="0"
                  link={
                    Object {
                      "title": "Coolblue",
                      "url": "/",
                    }
                  }
                />
              </div>
              <div
                key="1"
              >
                <ArrowPanel
                  image="cool-blue.svg"
                  key="1"
                  link={
                    Object {
                      "title": "Alternate",
                      "url": "/",
                    }
                  }
                />
              </div>
              <div
                key="2"
              >
                <ArrowPanel
                  image="cool-blue.svg"
                  key="2"
                  link={
                    Object {
                      "title": "Beddenreus",
                      "url": "/",
                    }
                  }
                />
              </div>
              <div
                key="3"
              >
                <ArrowPanel
                  image="cool-blue.svg"
                  key="3"
                  link={
                    Object {
                      "title": "De Bijenkorf",
                      "url": "/",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
